cmd/container-use/stdio_test.go

```diff
@@ -68,10 +81,35 @@ func NewMCPServerProcess(t *testing.T, testName string) *MCPServerProcess {
 	t.Cleanup(func() {
 		server.Close()
 	})
-
 	return server
 }
 
+func serverSandboxEnv(baseEnv []string) ([]string, string) {
+	root, err := os.MkdirTemp("", "cu-stdio-srv-")
+	if err != nil {
+		panic(err)
+	}
+	cache := filepath.Join(root, "cache")
+	tmp := filepath.Join(root, "tmp")
+	_ = os.MkdirAll(cache, 0o755)
+	_ = os.MkdirAll(tmp, 0o755)
+
+	env := append([]string{}, baseEnv...)
+	env = append(env, "XDG_CACHE_HOME="+cache)
+	if runtime.GOOS == "windows" {
+		env = append(env, "TEMP="+tmp, "TMP="+tmp)
+	} else {
+		env = append(env, "TMPDIR="+tmp)
+	}
+	return env, root
+}
```

Comment:
how is this different than the logic in TestMain? Are we providing additional per-server isolation? 

the intent of these stdio tests in particular is to surface failures that occur when multiple server processes share a single source repo AND fork. If I'm understanding this helper correctly, this change unshares their forks, and therefore this test will no longer repro the data races it's designed to repro?

or is CONTAINER_USE_CONFIG_DIR the only var that matters here?